src/main/java/org/example/config/DataSourceConfig.java

```diff
@@ -0,0 +1,13 @@
+package org.example.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class DataSourceConfig {
+
+    @Bean
+    public ConnectionConfig connectionConfig() {
+        return new ConnectionConfig();
+    }
```

Comment:
instead of manual configuration of Connection and choosing Driver manager use DataSource specific db bean:
```java
    @Bean
    public DataSource dataSource(){
        PGSimpleDataSource dataSource = new PGSimpleDataSource();
        dataSource.setURL(...);
        dataSource.setUser(...);
        dataSource.setPassword(...);
        return dataSource;
    }
```

Reply:
Changed from manual configuration of Connection to DataSource specific db bean